csvcubed/csvcubed/cli/error_mapping.py

```diff
@@ -0,0 +1,195 @@
+    MissingColumnDefinitionError:
+        "Column '{error[csv_column_title]}' is present in CSV but no configuration could be found.",
+    DuplicateColumnTitleError:
+        "There are multiple CSV columns with the title: '{error[csv_column_title]}'.",
+    ColumnValidationError:
+        (
+            "An error occurred when validating the column '{error[csv_column_title]}':" + linesep + 
+            "{error[error]}"
+        ),
+    ColumnNotFoundInDataError:
+        "Configuration found for column '{error[csv_column_title]}' but no corresponding column found in CSV.",
+    UnknownPydanticValidationError:
+        "An error was encountered when validating the cube. The error occurred in '{error[path]}' "
+        "and was reported as '{error[original_error]}'",
+    UndefinedMeasureUrisError:
+        "The Measure URI {error[undefined_values]} found in the data was not defined in the cube config.",
+    UndefinedUnitUrisError:
+        "The Unit URI {error[undefined_values]} found in the data was not defined in the cube config.",
+    UndefinedAttributeValueUrisError:
+        "The Attribute URI {error[undefined_values]} in column '{error[component][label]}' defined in the cube "
+        "config was not found in the data.",
+    ReservedUriValueError:
+        "The URI value(s) {error[conflicting_values]} conflict with the reserved value: '{error[reserved_identifier]}'.",
+    ConflictingUriSafeValuesError:
+        "A URI collision has been detected in: '{error[component]}'.",
+    ValidationError:
+        "A validation error occurred when validating the cube: '{error[message]}'."
+}
+
+
+def friendly_error_mapping(error: ValidationError) -> str:
+    """
+    Given an Exception / Error it returns an error message that is more user-friendly
+    """
+    try:
+        en = error.__class__.__name__
+        msg = MSG_TEMPLATE[error.__class__]
+        ed = error.as_dict()
+
+        if isinstance(error, SpecificValidationError):
+            error.get_error_url()
+
+        if isinstance(error, ConflictingUriSafeValuesError):
+            ed['component'] = str(ed['component_type']).split('.')[-2].capitalize()
+            for safe_uri, values in error.map_uri_safe_values_to_conflicting_labels.items():
+                msg += f"{linesep}The values {tuple(values)} all map to the same identifier '{safe_uri}'."
+
+        msg = msg.format(error=ed)
+
+        if isinstance(error, (MaxNumComponentsExceededError,
+                              MinNumComponentsNotSatisfiedError,
+                              WrongNumberComponentsError,
+                              NeitherDefinedError,
+                              IncompatibleComponentsError)
+                      ):
+            if error.additional_explanation:
+                msg += f"\nFurther details are: {error.additional_explanation}"
+
+        if isinstance(error, SpecificValidationError):
```

Comment:
Could you check that it's `HasErrorUrl` here instead of `SpecificValidationError`?

Reply:
Also, can you make sure to use `os.linesep` so we're more confident in our platform-independence.